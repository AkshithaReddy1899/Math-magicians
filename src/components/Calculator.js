import React from 'react';
import '../styles/calculator.css';
import Display from './Display';
import ButtonContainer from './ButtonContainer';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div className="calculator">
        <Display value="0" />
        <ButtonContainer />
      </div>
    );
  }
}

export default Calculator;
