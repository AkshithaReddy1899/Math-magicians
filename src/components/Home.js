import React from 'react';

function Home() {
  return (
    <div className="home">
      <h1>Welcome to our page!</h1>
      <div className="maindescription">
        <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Error magni itaque provi</p>
        <br />
        t atque consequuntur possimus quam, repellat at, soluta sunt maiores facere rerum
        <p>Lorem ipsum dolor sit amet consectetur adipisicing elit.</p>
      </div>
    </div>
  );
}

export default Home;
