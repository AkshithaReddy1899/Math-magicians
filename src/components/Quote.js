import React from 'react';

function Quote() {
  return (
    <div className="quote">
      <p>
        <span>
          Numbers have life; they are not just symbols on paper
        </span>
      </p>
      <b>__Shakuntala Devi.</b>
    </div>
  );
}

export default Quote;
