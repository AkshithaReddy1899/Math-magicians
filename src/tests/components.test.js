import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../components/Home';
import Calculator from '../components/Calculator';
import Quote from '../components/Quote';

describe('Test for components', () => {
  it('Home renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Calculator renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Quote renders correctly', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
