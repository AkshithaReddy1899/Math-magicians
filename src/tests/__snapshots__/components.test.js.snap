// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for components Calculator renders correctly 1`] = `
<div
  className="calculatorpage"
>
  <p>
    Let's do some math!
  </p>
  <div
    className="calculator"
  >
    <h1
      className="display"
    >
      0
    </h1>
    <div
      className="button-container"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
        value="AC"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
        value="+/-"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
        value="%"
      >
        %
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
        value="รท"
      >
        รท
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
        value="7"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
        value="8"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
        value="9"
      >
        9
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
        value="x"
      >
        x
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
        value="4"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
        value="5"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
        value="6"
      >
        6
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
        value="-"
      >
        -
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
        value="1"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
        value="2"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
        value="3"
      >
        3
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
        value="+"
      >
        +
      </button>
      <button
        className="zero"
        onClick={[Function]}
        type="button"
        value="0"
      >
        0
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
        value="."
      >
        .
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
        value="="
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Test for components Home renders correctly 1`] = `
<div
  className="home"
>
  <h1>
    Welcome to our page!
  </h1>
  <div
    className="maindescription"
  >
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Error magni itaque provi
    </p>
    <br />
    t atque consequuntur possimus quam, repellat at, soluta sunt maiores facere rerum
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </div>
</div>
`;

exports[`Test for components Quote renders correctly 1`] = `
<div
  className="quote"
>
  <p>
    <span>
      Numbers have life; they are not just symbols on paper
    </span>
  </p>
  <b>
    __Shakuntala Devi.
  </b>
</div>
`;
